packages/fluentui/react-bindings/src/icons/createSvgIcon.ts

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+// @ts-ignore
+import { ThemeContext } from 'react-fela';
+
+import { StylesContextValue } from '../styles/types';
+import useStyles from '../hooks/useStyles';
+import getUnhandledProps from '../utils/getUnhandledProps';
+import { SvgIconCreateFnParams, SvgIconProps } from './types';
+
+export const svgIconClassName = 'ui-icon';
+export const svgIconDisplayName = 'SvgIcon';
+
+export const svgIconHandledProps: (keyof SvgIconProps)[] = [
+  'alt',
+  'aria-label',
+  'bordered',
+  'className',
+  'circular',
+  'design',
+  'disabled',
+  'outline',
+  'size',
+  'rotate',
+  'styles',
+  'variables',
+  'xSpacing',
+];
+
+const createSvgIcon = <TProps = {}>({ svg, displayName, handledProps = [] }: SvgIconCreateFnParams<TProps>) => {
+  const Component: React.FC<React.HTMLAttributes<HTMLSpanElement> & TProps & SvgIconProps> & {
+    handledProps: (keyof (TProps & SvgIconProps))[];
+  } = props => {
+    const context: StylesContextValue = React.useContext(ThemeContext);
+
+    const {
+      alt,
+      'aria-label': ariaLabel,
+      bordered,
+      circular,
+      className,
+      design,
+      disabled,
+      outline,
+      rotate = 0,
+      size = 'medium',
+      styles,
+      variables,
+      xSpacing,
+    } = props;
+
+    const { classes } = useStyles(svgIconDisplayName, {
+      className: svgIconClassName,
+      mapPropsToStyles: () => ({
+        bordered,
+        circular,
+        disabled,
+        outline,
+        rotate,
+        size,
+        xSpacing,
+      }),
+      mapPropsToInlineStyles: () => ({ className, design, styles, variables }),
+      rtl: context.rtl,
+    });
+
+    const unhandledProps = getUnhandledProps([...svgIconHandledProps, ...handledProps], props);
+
+    return React.createElement(
+      'span',
+      {
+        role: 'img',
+        'aria-hidden': alt || ariaLabel ? undefined : 'true',
```

Comment:
@kolaps33 @jurokapsiar I copied these props from the iconBehavior, but as @silviuavram pointed out alt is not a valid span element and even more we are not applying it anywhere. Should we remove it?

Reply:
I brought it up after going through https://css-tricks.com/accessible-svgs/ .

There it says that in order to have `alt` like description in `<svg>` then the svg should have a child `<title>` with that text.
Also I see that they suggest adding `role="img"` on the `<svg>` element.

Everything is explained in the `2. Inline SVG` section. Let me know your thoughts.

Reply:
But as it worked before, we have the role="img" on the span element (based on the iconBeavior not the svg). If the role="img" should be on the svg, than I guess I should remove all attributes from here, and we can just add comments of best practices.

Reply:
BTW this means that the iconBehavior should probably be changed too.

Reply:
Resolved after offline discussion. We won't do any changes, as this is a working solution. If we decide to change the behavior in the future, we will update the factory too.